discojs/discojs-core/src/serialization/model.ts

```diff
@@ -22,45 +23,57 @@ export async function encode (model: Model): Promise<Encoded> {
   }
 
   if (model instanceof models.GPT) {
-    const serialized = await serialization.weights.encode(model.serialize())
-    return msgpack.encode([Type.GPT, serialized])
+    const { weights, config } = model.serialize()
+    const serializedWeights = await serialization.weights.encode(weights)
+    return msgpack.encode([Type.GPT, serializedWeights, config])
   }
 
   throw new Error('unknown model type')
 }
 
 export async function decode (encoded: unknown): Promise<Model> {
   if (!isEncoded(encoded)) {
-    throw new Error('invalid encoding')
+    throw new Error("Invalid encoding, raw encoding isn't an instance of Uint8Array")
   }
   const raw: unknown = msgpack.decode(encoded)
 
-  if (!Array.isArray(raw) || raw.length !== 2) {
-    throw new Error('invalid encoding')
+  if (!Array.isArray(raw) || raw.length < 2) {
+    throw new Error("invalid encoding, encoding isn't an array or doesn't contain enough values")
   }
-  const [type, rawModel] = raw as [unknown, unknown]
-
+  const type = raw[0] as unknown
   if (typeof type !== 'number') {
-    throw new Error('invalid encoding')
+    throw new Error('invalid encoding, first encoding field should be the model type')
   }
+  const rawModel = raw[1] as unknown
   switch (type) {
     case Type.TFJS:
+      if (raw.length !== 2) {
+        throw new Error('invalid encoding, TFJS model encoding should be an array of length 2')
+      }
       // TODO totally unsafe casting
       return await models.TFJS.deserialize(rawModel as tf.io.ModelArtifacts)
-    case Type.GPT: {
+    case Type.GPT: {  
+      let config
+      if (raw.length == 2) {
+        config = undefined
+      } else if (raw.length == 3) {
+        config = raw[2] as GPTConfig
```

Comment:
you don't check that it is indeed a valid `GPTConfig`. you can put a function `isGPTConfig` in `models/gpt/config` to check for that, inspired of the various checkers in `tasks/*`